include/rmm/mr/device/pool_memory_resource.hpp

```diff
@@ -51,6 +51,35 @@ namespace rmm::mr {
  * @{
  * @file
  */
+namespace detail {
+/**
+ * @brief A helper class to remove the device_accessible property
+ *
+ * We want to be able to use the pool_memory_resource with an upstream that may not
+ * be device accessible. To avoid rewriting the world, we allow to conditionally remove
+ * the cuda::mr::device_accessible property.
+ *
+ * @tparam PoolResource the pool_memory_resource class
+ * @tparam Upstream memory_resource to use for allocating the pool.
+ */
+template <class PoolResource, class Upstream, class = void>
+struct maybe_remove_property {};
+
+/**
+ * @brief Specialization of maybe_remove_property to not propagate non existing properties
+ */
+template <class PoolResource, class Upstream>
+struct maybe_remove_property<
+  PoolResource,
+  Upstream,
+  cuda::std::enable_if_t<!cuda::has_property<Upstream, cuda::mr::device_accessible>>> {
```

Comment:
nit: This is called `maybe_remove_property` but it is hardcoded to remove only `device_accessible`. Consider parameterizing the property it removes or rename to be more explicit.